presto-hive/src/main/java/com/facebook/presto/hive/HiveSplitManager.java

```diff
@@ -385,7 +389,11 @@ public CounterStat getHighMemorySplitSource()
             }
         }
 
-        Optional<HiveStorageFormat> storageFormat = getHiveStorageFormat(table.getStorage().getStorageFormat());
+        StorageFormat storageFormat = table.getStorage().getStorageFormat();
```

Comment:
shall we check session property or configuration here? according to the release note:
This is enabled when configuration property ``hive.parquet.use-column-names`` or the hive catalog session property ``parquet_use_column_names`` is set to true. By default they are mapped by index.

Reply:
@zhenxiao [Line 392](https://github.com/prestodb/presto/pull/16348/files/dc3c96ae0ae86a608bc03d6425b2651cac18e72a#diff-be0abdaf5165467c629fccc676e607333b40ac777c8987a7fdc1c3d52183811dR392) and [Line 393](https://github.com/prestodb/presto/pull/16348/files/dc3c96ae0ae86a608bc03d6425b2651cac18e72a#diff-be0abdaf5165467c629fccc676e607333b40ac777c8987a7fdc1c3d52183811dR393) are required in any case. Only lines [395](https://github.com/prestodb/presto/pull/16348/files/dc3c96ae0ae86a608bc03d6425b2651cac18e72a#diff-be0abdaf5165467c629fccc676e607333b40ac777c8987a7fdc1c3d52183811dR395) and [396](https://github.com/prestodb/presto/pull/16348/files/dc3c96ae0ae86a608bc03d6425b2651cac18e72a#diff-be0abdaf5165467c629fccc676e607333b40ac777c8987a7fdc1c3d52183811dR396) are not mandatory if `hive.parquet.use-column-names` is not set to true. But in any case value of this variable is only used in one method `getTableToPartitionMapping` which has the required check at [line 521](https://github.com/prestodb/presto/pull/16348/files/dc3c96ae0ae86a608bc03d6425b2651cac18e72a#diff-be0abdaf5165467c629fccc676e607333b40ac777c8987a7fdc1c3d52183811dR521). 

Anyways, I have added a check here as well, let me know if you feel it is not necessary.

Reply:
looks nice